config/v1/types_infrastructure.go

```diff
@@ -1558,30 +1561,6 @@ type EquinixMetalPlatformStatus struct {
 	IngressIP string `json:"ingressIP,omitempty"`
 }
 
-// PowervsServiceEndpoint stores the configuration of a custom url to
-// override existing defaults of PowerVS Services.
-type PowerVSServiceEndpoint struct {
```

Comment:
Where is this used? You cannot change the type of this, this would be a breaking change and would break serialisation for clients

Reply:
Looking at this again, the serialisation actually isn't an issue, the shape of the two APIs matches. The issue is the validations. 

You would be moving to tighter validation which could break existing users. If you want to move to a tighter validation you would have to create a ratcheting validation that allowed old values and then only enforced the tighter validation for changed values.

Looking at the URL, that's probably desirable.

For the name, are all of the IBMCloudServiceNames supported on Power? And what about if I specified Power for IBM VPC, what would happen?

Reply:
Looking into how to do ratcheting validation. If you have any pointers that would be appreciated!

Yes all IBMCloudServiceNames are supported on Power. Both providers exist in IBM Cloud and use many of the same services, which use the same endpoints. For example, when you override the IAM endpoint in Power VS, you wouldn't be overriding an endpoint unique to Power VS, you'd be overriding the same endpoint that IBM VPC would be using for IAM.

Overriding the Power endpoint when deploying IBM VPC would essentially be ignored. It would be valid, but IBM VPC never reaches out to the Power endpoint. Similarly, for Power VS, we could validly override the HyperProtect endpoint, but we'd never use it.

Hope that makes sense!

Reply:
Ratcheting validation means that you'll need to use CEL (for now at least) to validate the transition, with a `self == oldSelf || <rule>` style thing. 

The issue is that this doesn't always apply (on create it wouldn't work for example), but that is being ironed out upstream at the moment and should be fixed in a release or two IIUC

I'm hesitant at the moment to suggest we move forward here at all. There is of course a potential future where we want to add a field only to Power or only to VPC, in which case, we would want to split this out again. Is it worth the effort to unify this now, given that? Typically we advise to not re-use structs, so that the future evolution can happen without accidentally adding new fields in places we didn't want them in the first place

Reply:
@cjschaef @mkumatag @Karthik-K-N would appreciate any input you have!